app/models/post.rb

```diff
@@ -0,0 +1,27 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
+  has_many :comments
+  has_many :likes
+  has_many :users, through: :likes
+
+  first_user = User.find_by(name: 'Tom')
+
+  Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
+  Post.create(author: first_user, title: 'Hello', text: 'This is my second post')
+  Post.create(author: first_user, title: 'Hello', text: 'This is my third post')
+  Post.create(author: first_user, title: 'Hello', text: 'This is my fourth post')
+
+  Post.fine(1).delete
+
+  # A method that updates the posts counter for a user.
+  def self.update_post_counter(user_id)
+    user = User.find(user_id)
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [ Required ] I think it would be better if you refactor this method for the following reason since the method is already in the `Post` model and a `post` belongs to a `user` you don't have to pass the `user_id` to the `update_post_counter` method.

You can refactor the method as suggested below, this example uses a [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database.
## Example code
 ```ruby
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :comments
  has_many :likes

  after_save :update_post_counter
  
  def update_post_counter
    post.increment!(:post_counter)
  end
end
```